spec/src/main/doc/javaee/implementation_ee.asciidoc

```diff
@@ -0,0 +1,319 @@
+[[implementation_ee]]
+
+== Addition to programming model for Java EE
+
+When running in Java EE, the container must extend the rules defined in <<implementation>>, and must also provide built-in support for injection and contextual lifecycle management of the following kinds of bean:
+
+* Session beans
+* Resources (Java EE resources, persistence contexts, persistence units, remote EJBs and web services)
+
+Java EE and embeddable EJB containers are required by the Java EE and EJB specifications to support EJB session beans and the Java EE component environment. Other containers are not required to provide support for injection or lifecycle management of session beans or resources.
+
+[[what_classes_are_beans_ee]]
+
+==== Which Java classes are managed beans in Java EE?
+
+When running in Java EE, A top-level Java class is a managed bean if it meets requirement described in <<what_classes_are_beans>> and if it is defined to be a managed bean by any other Java EE specification and if
```

Comment:
this should be: "requirements" OR "defined to be a managed bean by any other Java EE specification", not AND